sidekick_vault/lib/src/gpg.dart

```diff
@@ -13,21 +13,44 @@ void requireGpg() {
   }
 }
 
-/// Returns the decrypted file located in a temp directory
-File gpgDecrypt(File file, String password) {
-  requireGpg();
-  final outDir = Directory.systemTemp.createTempSync();
-  final outFile = outDir.file(file.nameWithoutExtension);
+extension GpgEncrypt on File {
+  /// Encrypts the file using gpg.
+  void encrypt(String password) {
```

Comment:
I don't like this as extension. It doesn't make it easier to use

Reply:
I don't quite understand – why is it less adequate than any other extension method?

Reply:
Extensions are great because you can chain them. But this function returns `void`, chaining makes no sense.

Also, `File` is quite general, extensions should be avoided on common types such as primitives. `File` is commonly used but encrypt/decrypt is only useful for those located in the vault.

Extensions are still less supported than normal functions. Imports are still not working well, they don't appear in auto-completion.

I would only create an extension when it improves the API. This doesn't improve it, it just changes it. Therefore I prefer the function with better tooling support.

--- 

General advice: If you want your PR to be merged quickly - especially in open-source - try to avoid unnecessary discussions. 
- Create a minimal diffs
- Try to fit the code style that's already there
- Avoid refactoring when adding new features